src/deepsparse/yolov8/utils/validation/deepsparse_validator.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# flake8: noqa
+
+import json
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, List
+
+from tqdm import tqdm
+
+import torch
+from deepsparse import Pipeline
+from deepsparse.yolo import YOLOOutput
+from ultralytics.yolo.cfg import get_cfg
+from ultralytics.yolo.data.utils import check_det_dataset
+from ultralytics.yolo.engine.validator import BaseValidator
+from ultralytics.yolo.utils import (
+    DEFAULT_CFG,
+    LOGGER,
+    RANK,
+    SETTINGS,
+    TQDM_BAR_FORMAT,
+    callbacks,
+)
+from ultralytics.yolo.utils.files import increment_path
+from ultralytics.yolo.utils.ops import Profile
+from ultralytics.yolo.utils.torch_utils import select_device, smart_inference_mode
+
+
+__all__ = ["DeepSparseValidator"]
+
+
+def schema_to_tensor(pipeline_outputs: YOLOOutput, device: str) -> List[torch.Tensor]:
+    """
+    Transform the YOLOOutput to the format expected by the validation code.
+
+    :param pipeline_outputs: YOLOOutput from the pipeline
+    :param device: device to move the tensors to
+    :return list of tensor with the format [x1, y1, x2, y2, confidence, class]
+    """
+
+    preds = []
+
+    for boxes, labels, confidence in zip(
+        pipeline_outputs.boxes, pipeline_outputs.labels, pipeline_outputs.scores
+    ):
+
+        boxes = torch.tensor(boxes)
+
+        # map labels to integers and reshape for concatenation
+        labels = list(map(int, list(map(float, labels))))
+        labels = torch.tensor(labels).view(-1, 1)
+
+        # reshape for concatenation
+        scores = torch.tensor(confidence).view(-1, 1)
+        # concatenate and append to preds
+        preds.append(torch.cat([boxes, scores, labels], axis=1).to(device))
+    return preds
+
+
+# adapted from ULTRALYTICS GITHUB:
+# https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/engine/validator.py
+# the appropriate edits are marked with # deepsparse edit: <edit comment>
+
+
+class DeepSparseValidator(BaseValidator):  # deepsparse edit: overwriting BaseValidator
```

Comment:
Are the comments in a different format cause its adapted from ultralytcs? will sphinx pick this up?

Reply:
Yes, that's copypaste from ultralytics.